reference/alerts.v1.yaml

```diff
@@ -0,0 +1,168 @@
+openapi: 3.0.0
+info:
+  title: Alerts API
+  version: '1.0'
+  description: |-
+    The Tidepool API is an HTTP REST API used by Tidepool clients to communicate with the Tidepool Platform.
+
+    For more information, see the [Getting Started](../docs/quick-start.md) section.
+  termsOfService: https://developer.tidepool.org/terms-of-use/
+  contact:
+    name: API Support
+    url: https://support.tidepool.org/
+    email: support@tidepool.org
+  license:
+    name: BSD-2-Clause
+    url: https://github.com/tidepool-org/hydrophone/blob/master/LICENSE
+  x-tidepool-service: https://github.com/tidepool-org/hydrophone
+
+servers:
+  - url: 'https://external.integration.tidepool.org'
+    description: integration
+  - url: 'https://api.tidepool.org'
+    description: production
+  - url: 'https://dev1.dev.tidepool.org'
+    description: dev1
+  - url: 'https://qa1.development.tidepool.org'
+    description: qa1
+  - url: 'https://qa2.development.tidepool.org'
+    description: qa2
+
+security:
+  - sessionToken: []
+
+tags:
+  - name: Alerts
+    description: >-
+      Endpoints related to alert configuration and management.
+
+paths:
+  /alerts/{userId}/{followedUserId}:
```

Comment:
I don't have a full understanding of all that went into this, but consider a more RESTful endpoint, such as `/users/{userId}/followedUser/{followedUserId}/alerts`. Or maybe flip that around, `/users/{userId}/followers/{followerUserId}/alerts` where the user is the only being followed rather than the follower. Not sure which might make more sense here.

Reply:
> I don't have a full understanding of all that went into this, but consider a more RESTful endpoint, such as `/users/{userId}/followedUser/{followedUserId}/alerts`. Or maybe flip that around, `/users/{userId}/followers/{followerUserId}/alerts` where the user is the only being followed rather than the follower. Not sure which might make more sense here.

The code for this is already merged, but I guess not released, so if we're going to make changes we should do it now.

You obviously have stronger opinions on what RESTful means that I do, so please just let me know what you'd like to see, and I'll start on it.

Reply:
The code for this is approved, and things are being blocked. I'm going to merge the code and deploy it, we'll have to migrate/deprecate the endpoint later if we want to. It should be manageable as this endpoint should only be used within Tidepool anyway (that is, by Tidepool mobile clients). Though I guess there's nothing stopping a 3rd party from making a mobile client that would use it... It's marked internal anyway...

Reply:
@toddkazakov in discussions with @darinkrauss he prefers to defer to you with regard to whether the existing API path is acceptable, or if it should be changed to one of the suggestions he made above in https://github.com/tidepool-org/TidepoolApi/pull/89#discussion_r1473379277

I'm fine with any decision, and will make the changes requested, just, I guess let me know what needs to happen and I'll get on it.

Reply:
The suggestion by @darinkrauss is definitely a step in the right direction. If the endpoint is not used let's do the change. In the past I have been tripped up by the permissions API which has very similar style to the alerts endpoint.

Reply:
I'll implement `/users/{userId}/followers/{followerUserId}/alerts`

Reply:
I've gone with `/users/{followedUserID}/followers/{userID}/alerts`

This (hopefully) keeps the terminology in the handlers more consistent.

That is, in the handler, the user that's submitting the request is identified by `userID`, so when authorizing the request, we compare the userID path parameter to the userID from the authentication token, which seems very natural, as opposed to having to know that for this particular endpoint, one should compare some other user (followingUserID or followerUserID) to the token's userID.

Reply:
Well, I couldn't do the above, because one of our rest libraries won't let us have paths with common roots that use different names for the variables...